### ISSUE TITLE
Fresh UE4 Project Won't Build with spine-runtimes-3.7-beta-cpp

### ISSUE THREAD

**Issue by supersaiyansubtlety** [EsotericSoftware/spine-runtimes#1130]
I'm trying to start a 2D side scroller using using spine and UE4. Unreal Editor fails to build the project after following the instructions [here](http://esotericsoftware.com/spine-ue4). Building the project via Visual Studio 2015 yeilds the following errors: 
```
Severity	Code	Description	File	Line	Suppression State	Project
Error	C4723	potential divide by 0	f:\documents\unreal projects\bitty_bloody_unreal\plugins\spineplugin\source\spineplugin\public\spine-cpp\src\spine\mathutil.cpp	99		Bitty_Bloody_Unreal
Error	C4723	potential divide by 0	f:\documents\unreal projects\bitty_bloody_unreal\plugins\spineplugin\source\spineplugin\public\spine-cpp\src\spine\mathutil.cpp	99		Bitty_Bloody_Unreal
Error		UBT ERROR: Failed to produce item: F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Plugins\SpinePlugin\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor-SpinePlugin.lib	F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Intermediate\ProjectFiles\UnrealBuildTool	1		Bitty_Bloody_Unreal
Error	MSB3075	The command ""C:\Program Files\Epic Games\UE_4.19\Engine\Build\BatchFiles\Build.bat" Bitty_Bloody_UnrealEditor Win64 Development "F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Bitty_Bloody_Unreal.uproject" -WaitMutex -FromMsBuild" exited with code 5. Please verify that you have sufficient rights to run this command.	C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.MakeFile.Targets	41		Bitty_Bloody_Unreal
```
and this is the build output
```
1>------ Rebuild All started: Project: Bitty_Bloody_Unreal, Configuration: Development_Editor x64 ------
1>  Cleaning Bitty_Bloody_UnrealEditor Binaries...
1>  Creating makefile for Bitty_Bloody_UnrealEditor (no existing makefile)
1>  Performing full C++ include scan (no include cache file)
1>  Parsing headers for Bitty_Bloody_UnrealEditor
1>    Running UnrealHeaderTool "F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Bitty_Bloody_Unreal.uproject" "F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Intermediate\Build\Win64\Bitty_Bloody_UnrealEditor\Development\Bitty_Bloody_UnrealEditor.uhtmanifest" -LogCmds="loginit warning, logexit warning, logdatabase error" -Unattended -WarningsAsErrors -installed
1>  Reflection code generated for Bitty_Bloody_UnrealEditor in 11.6014426 seconds
1>  Performing 22 actions (12 in parallel)
1>  [5/22] Resource PCLaunch.rc
1>  [6/22] Resource PCLaunch.rc
1>  [4/22] Resource PCLaunch.rc
1>  [2/22] Resource ModuleVersionResource.rc.inl
1>  SharedPCH.Engine.cpp
1>  SharedPCH.UnrealEd.cpp
1>  Bitty_Bloody_UnrealCharacter.gen.cpp
1>  Module.SpinePlugin.cpp
1>  Module.SpinePlugin.gen.cpp
1>  Bitty_Bloody_Unreal.init.gen.cpp
1>  Bitty_Bloody_UnrealGameMode.gen.cpp
1>  Bitty_Bloody_UnrealCharacter.cpp
1>  Bitty_Bloody_UnrealGameMode.cpp
1>  Bitty_Bloody_Unreal.cpp
1>  [15/22] Link UE4Editor-Bitty_Bloody_Unreal.lib
1>     Creating library F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor-Bitty_Bloody_Unreal.lib and object F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor-Bitty_Bloody_Unreal.exp
1>  Module.SpineEditorPlugin.cpp
1>  Module.SpineEditorPlugin.gen.cpp
1>f:\documents\unreal projects\bitty_bloody_unreal\plugins\spineplugin\source\spineplugin\public\spine-cpp\src\spine\mathutil.cpp(99): error C4723: potential divide by 0
1>f:\documents\unreal projects\bitty_bloody_unreal\plugins\spineplugin\source\spineplugin\public\spine-cpp\src\spine\mathutil.cpp(99): error C4723: potential divide by 0
1>  [18/22] Link UE4Editor-SpineEditorPlugin.lib
1>     Creating library F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Plugins\SpinePlugin\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor-SpineEditorPlugin.lib and object F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Plugins\SpinePlugin\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor-SpineEditorPlugin.exp
1>UnrealBuildTool : error : UBT ERROR: Failed to produce item: F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Plugins\SpinePlugin\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor-SpinePlugin.lib
1>  Total build time: 120.35 seconds (Local executor: 0.00 seconds)
1>C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.MakeFile.Targets(46,5): error MSB3073: The command ""C:\Program Files\Epic Games\UE_4.19\Engine\Build\BatchFiles\Rebuild.bat" Bitty_Bloody_UnrealEditor Win64 Development "F:\Documents\Unreal Projects\Bitty_Bloody_Unreal\Bitty_Bloody_Unreal.uproject" -WaitMutex -FromMsBuild" exited with code -1.
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========
```

**Comment by badlogic** [comment]
Thank you, I missed a pragma for MSVC  in spine-cpp (see https://github.com/EsotericSoftware/spine-runtimes/commit/fd162c73a263c520c661013b0e953404a32962c6). Please update the 3.7-beta-cpp branch and copy the latest sources from spine-cpp to your UE4 project.

Thanks for reporting!

**Comment by supersaiyansubtlety** [comment]
Thank you! I finally got my project to build with the Spine plugin! However I've encountered another problem when trying to place my animation in the scene, I'll open another issue.
